pkg/engine/handlers/validation/utils.go

```diff
@@ -3,45 +3,32 @@ package validation
 import (
 	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
 	kyvernov2 "github.com/kyverno/kyverno/api/kyverno/v2"
-	kyvernov2beta1 "github.com/kyverno/kyverno/api/kyverno/v2beta1"
-	"github.com/kyverno/kyverno/pkg/utils/match"
+	engineutils "github.com/kyverno/kyverno/pkg/engine/utils"
+	authenticationv1 "k8s.io/api/authentication/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-func matchResource(resource unstructured.Unstructured, rule kyvernov1.Rule) bool {
-	if rule.MatchResources.All != nil || rule.MatchResources.Any != nil {
-		matched := match.CheckMatchesResources(
-			resource,
-			kyvernov2beta1.MatchResources{
-				Any: rule.MatchResources.Any,
-				All: rule.MatchResources.All,
-			},
-			make(map[string]string),
-			kyvernov2.RequestInfo{},
-			resource.GroupVersionKind(),
-			"",
-		)
-		if matched != nil {
-			return false
-		}
+func matchResource(resource unstructured.Unstructured, rule kyvernov1.Rule, namespaceLabels map[string]string, policyNamespace string, operation kyvernov1.AdmissionOperation) bool {
+	// cannot use admission info from the current request as the user can be different, if the rule matches on old request user info, it should skip
+	admissionInfo := kyvernov2.RequestInfo{
+		Roles:        []string{"kyverno:invalidrole"},
+		ClusterRoles: []string{"kyverno:invalidrole"},
+		AdmissionUserInfo: authenticationv1.UserInfo{
+			Username: "kyverno:kyverno-invalid-controller",
+			UID:      "kyverno:invaliduid",
+			Groups:   []string{"kyverno:invalidgroup"},
+		},
```

Comment:
Why do we need this?

Reply:
For old object verification, we cannot know which user created it, so if the rule requires userinfo to match, old object verification should be skipped

Example:
You want to block a user from updating a resource, so you match on that user and deny in all cases
What happens is that the new object verification fails, during old object verification, the userinfo will match and also result in fail
Causing the rule to skip